provisioning_client/tests/hsm_client_riot_ut/hsm_client_riot_ut.c

```diff
@@ -414,14 +500,20 @@ BEGIN_TEST_SUITE(hsm_client_riot_ut)
                 .IgnoreArgument_Pub()
                 .IgnoreArgument_Priv();
             STRICT_EXPECTED_CALL(DERtoPEM(IGNORED_PTR_ARG, ECC_PRIVATEKEY_TYPE, IGNORED_PTR_ARG, IGNORED_NUM_ARG));
+            // Skip the mbedtls_mpi_free calls (these are internal/non-exposed calls)
+            STRICT_EXPECTED_CALL(mbedtls_mpi_free(IGNORED_PTR_ARG)); // 25
+            STRICT_EXPECTED_CALL(mbedtls_mpi_free(IGNORED_PTR_ARG)); // 26
         }
 
 
-        STRICT_EXPECTED_CALL(DERInitContext(IGNORED_PTR_ARG, IGNORED_PTR_ARG, IGNORED_NUM_ARG)); // 24
-        STRICT_EXPECTED_CALL(X509GetDeviceCertTBS(IGNORED_PTR_ARG, IGNORED_PTR_ARG, IGNORED_NUM_ARG));
+        STRICT_EXPECTED_CALL(DERInitContext(IGNORED_PTR_ARG, IGNORED_PTR_ARG, IGNORED_NUM_ARG)); // 27
+        STRICT_EXPECTED_CALL(X509GetDeviceCertTBS(IGNORED_PTR_ARG, IGNORED_PTR_ARG, IGNORED_NUM_ARG, IGNORED_PTR_ARG, IGNORED_NUM_ARG));
         STRICT_EXPECTED_CALL(RiotCrypt_Sign(IGNORED_PTR_ARG, IGNORED_PTR_ARG, IGNORED_NUM_ARG, IGNORED_PTR_ARG));
         STRICT_EXPECTED_CALL(X509MakeDeviceCert(IGNORED_PTR_ARG, IGNORED_PTR_ARG));
         STRICT_EXPECTED_CALL(DERtoPEM(IGNORED_PTR_ARG, CERT_TYPE, IGNORED_PTR_ARG, IGNORED_NUM_ARG));
+        // Skip the mbedtls_mpi_free calls (these are internal/non-exposed calls)
+        STRICT_EXPECTED_CALL(mbedtls_mpi_free(IGNORED_PTR_ARG)); // 32
+        STRICT_EXPECTED_CALL(mbedtls_mpi_free(IGNORED_PTR_ARG)); // 33
```

Comment:
Its not code, its a comment I thought might be useful to leave in to explain why these are being skipped in the negative test, however I have no attachment to them and can remove if its vital.

Reply:
I think we're talking about different things: I'm talking about line# 426/518 through 430/522 (old file/new file) below.
It's not OK to have commented out code (I realize you didn't change that part but since you're editing the file, we can also clean-up.)

Before removing the code, I'd like to understand if the DER and DER2PEM convertors should have been called or not. (@jebrando do you remember why that code was originally added?)

Reply:
Shouldn't be needed in the context of this UT as we are creating a root signed device cert based on the a simulation/sample derived root ca